app/models/condition.rb

```diff
@@ -0,0 +1,114 @@
+require_relative "date_formater"
+
+class Condition < ActiveRecord::Base
+  include DateFormat
+  has_many :trips, class_name: "Trip", primary_key: "date", foreign_key: "start_date"
+
+  validates :date,
+            :max_temperature_f,
+            :mean_temperature_f,
+            :min_temperature_f,
+            :mean_humidity,
+            :mean_visibility_miles,
+            :max_wind_speed_mph,
+            :precipitation_inches,
+              presence: true
+
+  def self.write(condition_details)
+    find_or_create_by(condition_details)
+  end
+
+  def self.days_with_high_temps_in(range)
+    where(max_temperature_f: range)
+  end
+
+  def self.grouped_by_date_number_of_trips
+    joins(:trips).group(:date).count("id")
+  end
+
+  def self.grouped_days_in_range_max_temp(range)
+    days_with_high_temps_in(range).joins(:trips).group(:date).count("id")
+  end
+
+  def self.average_number_of_rides_in_range_max_temp(range)
```

Comment:
Thank you @case-eee it is clear now. Tagging @MarisaMBurton @wlffann and @Sh1pley to make sure they see this thread.